samples/enumdevdir.c

```diff
@@ -16,31 +16,47 @@
 #include <cc65.h>
 
 
-void printdir (char *newdir)
+int printdir (char *newdir)
 {
-    char olddir[FILENAME_MAX];
-    char curdir[FILENAME_MAX];
+    char *olddir;
+    char *curdir;
     DIR *dir;
     struct dirent *ent;
     char *subdirs = NULL;
     unsigned dirnum = 0;
     unsigned num;
 
-    getcwd (olddir, sizeof (olddir));
+    olddir = malloc (FILENAME_MAX);
+    if (olddir != NULL) {
```

Comment:
This seems indeed untested. As far as I know it needs to be `==` instead of `!=`.

Reply:
Stupid...